### DISCUSSION TITLE
LocalNotification is not removing when application completely close.

### DISCUSSION THREAD

**Discussion by bangaras** [thudugala/Plugin.LocalNotification#460] [Q&A] [answered]
Hello, I am new to .NET MAUI Blazor,
I have tried this all method still my notification shows when app completely close/destroy. Also not getting that when my app completely closes then it executing OnDisappearing and OnDestroy from MainActivity or not. Please check

LocalNotificationCenter.Current.Cancel();
LocalNotificationCenter.Current.CancelAll();
LocalNotificationCenter.Current.Clear();
LocalNotificationCenter.Current.ClearAll();
 

//MainPage.xaml.cs 

```
protected override void OnDisappearing()
    {
        LocalNotificationCenter.Current.CancelAll();

        base.OnDisappearing();
    }
```

//MainActivity.cs

```
    protected override void OnDestroy()
    {
        LocalNotificationCenter.Current.CancelAll();

        base.OnDestroy();
    }
```

**Comment by thudugala** [comment, marked as answer]
@bangaras please report an issue for this a attach a sample project

**Comment by bangaras** [comment]
[MauiApp3.zip](https://github.com/thudugala/Plugin.LocalNotification/files/14447413/MauiApp3.zip)

@thudugala 
I have attached a sample project. Please Test/check in Android

I have added codes in these files
Pages\Counter.razor
MauiProgram.cs
MainPage.xaml.cs
Platforms\Android\MainActivity.cs

Steps for Reproduce issue:
1. Open app in Android device or in Android Emulator 
2. Turn on the notification permission 
3. Click on Enable Notification button, it will show notification
4. Then close/kill the app completely from recent
5. You will still see notification on top. 

https://github.com/thudugala/Plugin.LocalNotification/assets/72083437/1a28a703-8321-4819-aa4d-e4ac1559e946

**Reply by thudugala** [reply]
Hi @bangaras I will check. But I assume in android you cannot clean already fired notifications.

Can you clear an already fired notification using cancel , if the app was running in the foreground?

**Reply by bangaras** [reply]
Hello @thudugala, I can clear fired notification when app is running with 
`
 LocalNotificationCenter.Current.CancelAll();
 LocalNotificationCenter.Current.ClearAll();
`

But when I close the app completely that time I am not able to execute this CancelAll or ClearAll methods. I set these methods on 'OnDestory', 'OnDisappearing' events but its not executing when app closes completely. 

Can you suggest me any different way or any solution for clearing notification when app closes completely.
